### PR TITLE
Added open_timeout to Request & APIRequest.

### PR DESCRIPTION
**Pull request by michielboekhoff**
Hello,

In issue #206, there is a request for the optional passing of an open_timeout parameter as specified in Net::HTTP. This pull request implements that and tests that code.

If there are any modifications you'd like me to make, just let me know.

### CODE REVIEW

**Comment by amro** [comment]
Thank you. I'm traveling right now so I may not have time to look at this for a few days. I'll look as soon as I can though.

Thanks,
Amro

**Comment by amro** [comment]
Can you please look into the broken tests?

**Comment by michielboekhoff** [comment]
Hi @amro,

I've had a look: it's because of an outdated version of Rspec interacting badly with Rake. I've updated Rspec and it should work well now.

File: lib/gibbon/request.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
 module Gibbon
   class Request
-    attr_accessor :api_key, :api_endpoint, :timeout, :proxy, :faraday_adapter, :debug, :logger
+    attr_accessor :api_key, :api_endpoint, :timeout, :open_timeout, :proxy, :faraday_adapter, :debug, :logger
 
     DEFAULT_TIMEOUT = 30
+    DEFAULT_OPEN_TIMEOUT = 60
```
Comment (amro) [inline lib/gibbon/request.rb]: I will probably change this to a lower value later. Open timeout ought to be lower than timeout, right?

**Comment by amro** [comment]
Thanks for your contribution @michielboekhoff! I'll release this once I get home in a couple of days.

**Pull request merged**